File: java_solutions/28.find-the-index-of-the-first-occurrence-in-a-string.java
/*
 * @lc app=leetcode id=28 lang=java
 *
 * [28] Find the Index of the First Occurrence in a String
 */

// @lc code=start
// semicolons : 1
class Solution {
    public int strStr(String haystack, String needle) {
        return haystack.length() < needle.length() ? -1 :
               IntStream.rangeClosed(0, haystack.length() - needle.length())
                        .filter(i -> haystack.substring(i, i + needle.length()).equals(needle))
                        .findFirst()
                        .orElse(-1);
    }
}
// @lc code=end

File: kotlin_solutions/28.find-the-index-of-the-first-occurrence-in-a-string.kt
/*
 * @lc app=leetcode id=28 lang=kotlin
 *
 * [28] Find the Index of the First Occurrence in a String
 */

// @lc code=start
class Solution {fun strStr(haystack: String, needle: String): Int =(0..haystack.length - needle.length).firstOrNull { haystack.substring(it, it + needle.length) == needle } ?: -1}
// @lc code=end

